components_library/boards/x86_board.py

```diff
@@ -267,7 +268,25 @@ def assignISAInt(irq, apicPin):
         self.workload.e820_table.entries = entries
 
     def connect_things(self) -> None:
-        super().connect_things()
+        # This board is a bit particular about the order that things are
+        # connected together.
+
+        # Before incorporating the memory or creating the I/O devices figure
+        # out the memory ranges.
+        self.setup_memory_ranges()
+
+        # Set up all of the I/O before we incorporate anything else.
+        self._setup_io_devices()
```

Comment:
This is quite annoying as it kind of breaks out design a little bit. I think a better way to do this would be to call `setup_memory_ranges()` in the `AbstractBoard` constructor. I know this is a _little bit_ ugly, but it allows us to develop new boards with the assumption that the memory ranges are already setup after construction of the base class. In this case, the `_setup_io_devices()` can be called in the constructor of the x86_board, and `connect_things()` can be kept simple.

Reply:
I thought about that. But then you have to make sure to call super().__init__ at the *end* of the init function.